test/src/unit-max-tile-size.cc

```diff
@@ -0,0 +1,1116 @@
+    // when all validity values are zero. dim_data should always contain data.
+    attr_data.reserve(dim_data.size());
+
+    // Prepare offsets to attr items...
+    std::vector<uint64_t> attr_offsets(dim_data.size(), 0);
+    // ...offsets for those for which actually have data;
+    for (decltype(attr_data.size()) i = 0; i < attr_data.size(); i++) {
+      attr_offsets[i] = i;
+    }
+    // ...pad out offsets for those which do not have data.
+    for (decltype(attr_data.size()) i = attr_data.size(); i < dim_data.size();
+         ++i) {
+      attr_offsets[i] = attr_data.size();
+    }
+
+    // Prepare the validity values for the data
+    std::vector<uint8_t> attr_val(dim_data.size());
+    // attr_val also needs a non empty buffer, and ...
+    attr_val.reserve(dim_data.size());
+    // ... init those for which there is data
+    for (decltype(attr_data.size()) i = 0; i < attr_data.size(); ++i) {
+      attr_val[i] = 1;
+    }
+    // ... init those trailing which are empty and have no data
+    for (decltype(attr_data.size()) i = attr_data.size(); i < dim_data.size();
+         ++i) {
+      attr_val[i] = 0;
+    }
+
+    // Open the array for writing and create the query.
+    tiledb::Array array(ctx_, main_array_name_, TILEDB_WRITE);
+    tiledb::Query query(ctx_, array);
+    query.set_layout(TILEDB_UNORDERED)
+        .set_data_buffer("a1", attr_data)
+        .set_offsets_buffer("a1", attr_offsets)
+        .set_validity_buffer("a1", attr_val)
+        .set_data_buffer("d1", dim_data)
+        .set_offsets_buffer("d1", dim_offsets);
+
+    // Perform the write and close the array.
+    query.submit();
+    array.close();
+  }
+
+  uint64_t c_get_fragments_max_in_memory_tile_size(
+      const std::string& array_uri) {
+    uint64_t max_in_memory_tile_size = 0;
+    CHECK(
+        tiledb_array_maximum_tile_size(
+            &*ctx_.ptr(),
+            array_uri.c_str(),
+            &max_in_memory_tile_size, 
+            nullptr) == TILEDB_OK);
+
+    if (showing_data_) {
+      std::cout << "maximum in memory tile size "
+                << max_in_memory_tile_size
+                << std::endl;
+    }
+    return max_in_memory_tile_size;
+  }
+
+  uint64_t cpp_get_fragments_max_in_memory_tile_size(const std::string& array_uri) {
+    tiledb::Array array(ctx_, array_uri, TILEDB_READ);
+    uint64_t max_in_memory_tile_size = 0;
+    array.get_max_in_memory_tile_size(&max_in_memory_tile_size);
+
+    if (showing_data_) {
+      std::cout << "maximum in memory tile size " << max_in_memory_tile_size
+                << std::endl;
+    }
+    return max_in_memory_tile_size;
+  }
+
+  tiledb::Context ctx_;
+  tiledb::VFS vfs_;
+};  // struct MaxTileSizeFx
+
+TEST_CASE_METHOD(
+    MaxTileSizeFx,
+    "C++ API: Max tile size, dense, fixed, with consolidation",
+    "[capi][cppapi][max-tile-size][dense][consolidate]") {
+  remove_temp_dir(main_array_name_.c_str());
```

Comment:
This could be made part of the destructor of the test fixture and removed everywhere.

Reply:
No, tests will fail if that is done as array creates do not like seeing an array already existing, and all of the arrays use the same name.
So, the delete must be done at some point (beginning or end) in the test case.

I prefer deleting at the beginning (doesn't fail if doesn't exist).
It allows
A) a restart/repeat of a test without manual intervention if the array was left from a previous (possible aborted, or not) run.
B) examining the array after a test has finished/aborted
C) the serial execution of test cases using the same array name
If it is removed without being at least placed at the end of each test, subsequent tests will not run.
I have consistified its use, moving it out of the test cases and into the beginning of each of the create...array() routines.